### PR TITLE
Terminal Tool

### PR DESCRIPTION
**Pull request by Jeadie**
# 🗣 Description
```yaml
models:
- name: 4o-mini
    from: openai:gpt-4o-mini
    params:
      openai_api_key: ${ secrets:SPICE_OPENAI_API_KEY }
      spice_tools: iterm

tools:
  - name: iterm
    from: extensions:terminal
```

- Extension tools are not bundled into `spice_tools: auto`.

## 📄 Documentation Updates
- [ ] Documentation for terminal.

### CODE REVIEW

File: crates/runtime/src/tools/extension/terminal/manager.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    /// Spawns a new terminal session and adds it to the manager.
+    ///
+    /// # Returns
+    ///
+    /// * `io::Result<usize>` - The unique ID of the new terminal or an I/O error.
+    pub async fn spawn_terminal(&self) -> io::Result<usize> {
+        // Generate a unique ID for the new terminal
+        let mut id_lock = self.next_id.lock().await;
```
Comment (sgrebnov) [inline crates/runtime/src/tools/extension/terminal/manager.rs:58]: Nit: alternatively can be AtomicUsize that supports thread safe incremental updates

**Review by sgrebnov — APPROVED** [review]

**Comment by sgrebnov** [comment]
Tools are fine to include IMO as they are NOT included in default build, require special compile feature flag

File: bin/spiced/Cargo.toml
Comment (phillipleblanc) [inline bin/spiced/Cargo.toml:1]: This should have a proper enhancement with spec/security review before merging

**Comment by Jeadie** [comment]
Closing, to be included in: #4015

**Pull request closed without merging**